### DISCUSSION TITLE
asctime gives None in logging

### DISCUSSION THREAD

**Discussion by Tico06** [micropython/micropython#13719] [Libraries & Drivers] [answered]
Hi There,
I am using logging from standard-lib and trying to format the output with the following:
```
logging.basicConfig(format='%(asctime)s:%(levelname)-8s:%(message)s',
                    level=logging.DEBUG)
```

It results with the following:
```
mpremote rtc --set + mount . + run testing_logging.py
Local directory . is mounted at /remote
None:DEBUG   :My message.
```
How could I get the timestamp heading log line ?
Didn't find any working sample with this.

Thanks and regards,
Eric.

**Comment by rkompass** [comment, marked as answer]
There is a time module in standard-lib and the logging module needs it.
So first importing that module should solve your problem.
I had this problem once too and started to code `time.strftime() ` only to note later that it was already there.

> **time**
> This library extends the built-in [MicroPython time module](https://docs.micropython.org/en/latest/library/time.html#module-time) to include [time.strftime()](https://docs.python.org/3/library/datetime.html#strftime-and-strptime-behavior).

**Reply by Tico06** [reply]
Hi rkompass,

Thanks for your quick answer.
I imported this time extension in logging.py and it works like a charm. Is it possible to import outside ? I don't think.

Shouldn't be considered as a bug ? At least, it should be documented.

Thanks and reg,

Eric.

**Reply by rkompass** [reply]
Feel free to raise an issue (or documentation PR?). It should be documented, I agree.
And so it may save others some trouble.

**Reply by cahudson** [reply]
FYI - to provide more information on this. If you are running into this issue you need to download from here: [https://github.com/micropython/micropython-lib/blob/master/python-stdlib/time/time.py](https://github.com/micropython/micropython-lib/blob/master/python-stdlib/time/time.py) the time.py and add it into the folder with your logging.py file.[](url)
